cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```diff
@@ -17,6 +21,9 @@ metadata:
 spec:
 {{- if .Values.kubecostDeployment }}
   replicas: {{ .Values.kubecostDeployment.replicas | default 1 }}
+{{- end }}
+{{- if and .Values.kubecostDeployment.statefulset.enabled .Values.kubecostDeployment.leaderFollower.enabled }}
+  serviceName: {{ template "cost-analyzer.serviceName" . }}
```

Comment:
Can you help me understand why a servicename field gets added if we add a statefulset?

Reply:
A StatefulSet needs a `serviceName` by definition https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/